### DISCUSSION TITLE
Possibility to not set the interface type for esx host if interface is child interface

### DISCUSSION THREAD

**Discussion by dschartz** [bb-Ricardo/netbox-sync#323] [Q&A]
Hi, we are using Npar on our physical interfaces, So i define the Vmnic interfaces as child interfaces of our physically connected interfaces.

So for the moment the updates of the interfaces fails as setting the interface type is only permitted on non-child interfaces.
I there any possibility to check if interface pas a parent interface and ommit the Interface type change ? if not could you point me to the right spot to implement this check eventually ?

By the way, this is really an awesome tool

**Comment by bb-Ricardo** [comment]
Hi,

Sorry for not getting back earlier. I somehow missed it in my messages.

Currently this is not implemented and I'm unsure if I understood the whole context.

Would you be able to run the script in `-l DEBUG3` mode and redirect all the output to a file?

I would need just the vCenter API response of one host with this setup and then I could look into it. I don't have acces to an ESXi with this setup, sorry.

**Comment by dschartz** [comment]
Hi,Thanks for your response,In my config i have physical interfaces which are virtualized into 8 virual interfaces which are presented to Vmware esxi Host as vmnic0-8. As only the physical onterfaces are connected, i linked vmnic 0-8 to the physical interface as child interfaceSince your script wants to set th interface type 10Gbit for example, this config generates error sonce only physical interfaces can be assigned interface types like 10gbit sfp.So it would be cool if there was a setting which makes that the interface type is not set. This would allow to assign vmnic0-x as virtual interface and give a correct view of which vmnic  interface passes through which physical interface. I don’t talk bout discovering the relationship, just not serting the interface type which is incomatible with a virtual interfaceTo test this you can create a physical interface (eth0) create some virtual interfaces (vmnic0-x) as child interfaces to eth0 and let your script run on the esxi host objectBest regardsDaniel SchartzOn 14 Jul 2023, at 16:57, Ricardo ***@***.***> wrote:﻿
Hi,
Sorry for not getting back earlier. I somehow missed it in my messages.
Currently this is not implemented and I'm unsure if I understood the whole context.
Would you be able to run the script in -l DEBUG3 mode and redirect all the output to a file?
I would need just the vCenter API response of one host with this setup and then I could look into it. I don't have acces to an ESXi with this setup, sorry.

**Comment by bb-Ricardo** [comment]
Ah, so you manually assign the virtual interfaces to physical interfaces. And then the script tries to update the interface information and fails as it's not permitted to set interface speed for an virtual interface.

Let's see if I can mitigate this issue.

**Comment by dschartz** [comment]
That woud be super,If you check at the existence of the vmnic interface if type is virtual omit the setting of the type should do the trickLGDaniel SchartzOn 17 Jul 2023, at 10:24, Ricardo ***@***.***> wrote:﻿
Ah, so you manually assign the virtual interfaces to physical interfaces. And then the script tries to update the interface information and fails as it's not permitted to set interface speed for an virtual interface.
Let's see if I can mitigate this issue.
